server/lib/project/test/projectManager.test.js

```diff
@@ -258,12 +260,14 @@ describe('ProjectManager tests', function() {
 
   it('ACCEPT an OPEN project - change to PRODUCTION', function* () {
     const projectArgs = createProjectArgs(util.uid());
+    var sleep = require('thread-sleep');
     // create project
     yield contract.createProject(projectArgs);
     // set the state
     const newState = yield contract.handleEvent(projectArgs.name, ProjectEvent.ACCEPT);
     assert.equal(newState, ProjectState.PRODUCTION, 'handleEvent should return ProjectState.PRODUCTION');
     // check the new state
+    var res = sleep(8000);
```

Comment:
time based delays never guarantee the success of the operation.
is there an indicator we can query that will guarantee data integrity ?
you can see how we implemented the Searchable interface in `blockapps-sol / EChO`

Reply:
Same